### PR TITLE
Tnp core 100

### PR DESCRIPTION
**Pull request by SteynSean11**
### **PR Type**
Enhancement, Tests

___

### **Description**
- Implement raw data persistence layer with PostgreSQL before processing
  - Add Alembic migrations and RawIngestion model for durable data lake
  - Persist raw payloads atomically in all three ingestion endpoints (text/file/repo)
  - Ensure 100% data retention with duplicate detection (HTTP 409)

- Refactor conversation processing to use OmegaKG validation API gateway
  - Replace direct Neo4j writes with HTTP client for structured knowledge digests
  - Implement exponential backoff retry logic and error handling
  - Add KnowledgeDigest model for standardized API communication

- Add comprehensive integration tests and database session management
  - Create sync/async session factories for raw persistence and background processing
  - Add 305-line integration test suite covering persistence, recovery, and worker polling
  - Verify crash recovery and processing attempt tracking

- Configure OmegaKG integration and update dependencies
  - Add Alembic, SQLAlchemy, psycopg2, and httpx to dependencies
  - Add OmegaKG API URL and authentication token configuration
  - Remove Neo4j dependency in favor of API gateway pattern

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Ingestion Endpoints<br/>text/file/repo"] -->|"1. Persist raw data"| B["PostgreSQL<br/>raw_ingestions table"]
  B -->|"2. Return 202/200"| C["HTTP Response"]
  B -->|"3. Worker polls"| D["Background Processing"]
  D -->|"4. Create digest"| E["KnowledgeDigest"]
  E -->|"5. Validate & store"| F["OmegaKG API<br/>Gateway"]
  F -->|"6. Update status"| B
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td><strong>env.py</strong><dd><code>Configure Alembic environment with application settings</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-cd4d26a53ff97e019b612811dfa604ba24d5dfd2f4dc1b195f3e74d85c38e630">+108/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>alembic.ini</strong><dd><code>Alembic configuration file for migration management</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-91e6e909c6988f5ced15e45b2be3123a9dfacbeffbe781083f3c24d81069df12">+150/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>script.py.mako</strong><dd><code>Alembic migration template for new revisions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-f205a44c5aa9f79c6aa297e02809df6db77befcf48c6c9661d5daa2854e79f2c">+28/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>config.py</strong><dd><code>Add OmegaKG API configuration and authentication tokens</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-43a9865fb0a3d6f51d3efde996f509c78782dcf819cc44fc9889adf11df31d5e">+16/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Database schema</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>1c6ce4b88374_create_raw_ingestions.py</strong><dd><code>Create raw_ingestions table with indexes for worker polling</code></dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-b8d8a0af94fddfbdef78cad0b46da4d49a91d9b8eb2afd81ad3418386251a0da">+54/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>README</strong><dd><code>Alembic documentation placeholder</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-8ed52cd48b623cb510db84f21ba82f6e78a573e439490d6f9afa02d0981f61a0">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>.env.example</strong><dd><code>Add OmegaKG API configuration environment variables</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8c">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>10 files</summary><table>
<tr>
  <td><strong>__init__.py</strong><dd><code>Database models package initialization</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-83e72eb1376702c5dde776f8fa02b0511b36ea861c38c5f1c0081ec8f3b9041b">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>base.py</strong><dd><code>SQLAlchemy declarative base for ORM models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-61182eb3e50ca5c54cd2dd84dd7d01976d3863b68ab0b1e3ca4e0d813c488d22">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>raw_ingestion.py</strong><dd><code>RawIngestion model for durable data lake persistence</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-372f9fe4bdf3bf06e1a76dc7e17efc328847045555b95edc56b32090fb8c976a">+53/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>__init__.py</strong><dd><code>Database package initialization with session exports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-29530e79be6afdd4cb5841c33b06762195e4980ef5e2ee7b7dac7a43a1fc98eb">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>session.py</strong><dd><code>Sync/async database session factories for raw persistence</code></dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-9770e452668ee02b00cf560c5b714769ac3a935dab8262660c1a37b35b13b543">+105/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>ingestion.py</strong><dd><code>Add raw persistence to text and file ingestion endpoints</code>&nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-36985965a5a1d2da155f16ca9818ff0c4fae7f163663b4ee42f2e1941cbca43b">+106/-1</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>repository.py</strong><dd><code>Add raw persistence to python-repo ingestion endpoint</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-87bf74ee0fb4087c78f25ecd0ec2a679e77c5617790e47729712c1e207af9475">+49/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>knowledge_digest.py</strong><dd><code>KnowledgeDigest model for OmegaKG validation API</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-b21eb5b9ba47d648bfa635a3e1f79dc132cd841cf0cfca0488d3a1ebd76a141b">+70/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>omegakg_client.py</strong><dd><code>HTTP client for OmegaKG validation API with retry logic</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-1435dbb781f005290aa9651f5336df09b1fd52ff3a57d672add07b0cda29bbcd">+130/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>conversation_ingestor.py</strong><dd><code>Refactor to use OmegaKG API instead of direct Neo4j writes</code></dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-c4a1bd48f90bb427a2cfeeae6fd52937034f1f0b3d273591194ec7202273862a">+65/-40</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>neo4j_service.py</strong><dd><code>Remove direct Neo4j service in favor of API gateway</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-45a00555f2fec0c4df4242a53d81b563e0866064b6787c0fe52908d6a21bdbe6">+0/-85</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>test_raw_persistence.py</strong><dd><code>Comprehensive integration tests for raw persistence layer</code></dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-58717ec1cb68924f4ca58ff50cd85ff8725324d3dc9f410fcba374d75083fac1">+305/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_tn_core_103.py</strong><dd><code>Verification tests for OmegaKG integration refactor</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-04258a1f85ae290b50d01d3fe784f3bb32b88749a805f5f74ac77fdfdb423cf4">+80/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>pyproject.toml</strong><dd><code>Update dependencies for database and API client support</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711">+4/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>__init__.py</strong></td>
  <td><a href="https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-048657e1448659e68bfaf1ae34ef18ab2f722a693c21d2069d08b3d8dee533f7">[link]</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by SteynSean11** [comment]
> ## PR Compliance Guide 🔍
> Below is a summary of compliance checks for this PR:
> 
> **Security Compliance**
> ⚪	
> **Resource exhaustion **
> 
> Description: The `/ingest/text` endpoint persists `request.model_dump()` to `raw_payload` without any size/complexity limits, enabling an attacker to submit extremely large JSON bodies that can exhaust DB/storage and degrade availability (resource-exhaustion/DoS). **[ingestion.py [103-138]](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-36985965a5a1d2da155f16ca9818ff0c4fae7f163663b4ee42f2e1941cbca43bR103-R138)**
> 
> Referred Code
> ```python
> # STEP 1: IMMEDIATE RAW PERSISTENCE (TN-CORE-101)
> # Write raw data to PostgreSQL BEFORE any processing to ensure data retention
> try:
>     raw_record = RawIngestion(
>         ingestion_id=ingestion_id,
>         source_type="text",
>         content_type=request.metadata.content_type.value,
>         raw_payload=request.model_dump(mode="json"),
>         raw_metadata={
>             "source": request.metadata.source.value,
>             "tags": request.metadata.tags,
>             "source_url": request.metadata.source_url,
>             "title": request.metadata.title,
>         },
>         captured_at=datetime.utcnow(),
>         processed=False,
>     )
>     db.add(raw_record)
>     db.commit()
> 
>     logger.info(f"Raw ingestion persisted: {ingestion_id}")
> 
> 
>  ... (clipped 15 lines)
> ```
> 
> **Insecure token transport **
> 
> Description: The client uses `settings.omegakg_api_url` (defaulting to `http://localhost:8765`) and will send `Authorization: Bearer ...` over non-TLS HTTP if configured similarly in non-local environments, risking token disclosure via network interception. **[omegakg_client.py [21-37]](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-1435dbb781f005290aa9651f5336df09b1fd52ff3a57d672add07b0cda29bbcdR21-R37)**
> 
> Referred Code
> ```python
> def __init__(self):
>     self.settings = get_settings()
>     self.base_url = self.settings.omegakg_api_url
>     self.token = self.settings.bws_access_token or self.settings.static_service_token
>     self.timeout = httpx.Timeout(30.0, connect=5.0)
> 
>     headers = {
>         "Content-Type": "application/json",
>     }
>     if self.token:
>         headers["Authorization"] = f"Bearer {self.token}"
> 
>     self.client = httpx.AsyncClient(
>         base_url=self.base_url,
>         timeout=self.timeout,
>         headers=headers
>     )
> ```
> 
> **Hardcoded token example **
> 
> Description: The example configuration includes a realistic-looking `STATIC_SERVICE_TOKEN=dev-token-12345`, which may be copied into real deployments and accidentally reused/committed, increasing the risk of credential exposure and unauthorized access. **[.env.example [224-232]](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR224-R232)**
> 
> Referred Code
> ```
> # =============================================================================
> # OmegaKG INTEGRATION (TN-CORE-103)
> # =============================================================================
> # Base URL for the OmegaKG validation API
> INGEST_OMEGAKG_API_URL=http://localhost:8765
> # Service authentication token (Zero Trust)
> BWS_ACCESS_TOKEN=<your_bws_token>
> STATIC_SERVICE_TOKEN=dev-token-12345
> ```
> 
> **Ticket Compliance**
> ⚪	
> 🎫 **No ticket provided **
> * [ ]  Create ticket/issue
> 
> **Codebase Duplication Compliance**
> ⚪	
> **Codebase context is not defined **
> Follow the [guide](https://qodo-merge-docs.qodo.ai/core-abilities/rag_context_enrichment/) to enable codebase context checks.
> 
> **Custom Compliance**
> 🔴	
> **Generic: Meaningful Naming and Self-Documenting Code**
> 
> **Objective:** Ensure all identifiers clearly express their purpose and intent, making code self-documenting
> 
> **Status:** [**Wrong attribute name**](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-58717ec1cb68924f4ca58ff50cd85ff8725324d3dc9f410fcba374d75083fac1R74-R75): The new tests reference `raw_record.metadata` even though the model field is `raw_metadata`, which is misleading and likely incorrect.
> 
> Referred Code
> ```python
> assert raw_record.metadata["tags"] == ["test", "python"]
> assert raw_record.processed == False
> ```
> 
> > Learn more about managing compliance [generic rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#configuration-options) or creating your own [custom rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#custom-compliance)
> 
> **Generic: Robust Error Handling and Edge Case Management**
> 
> **Objective:** Ensure comprehensive error handling that provides meaningful context and graceful degradation
> 
> **Status:** [**Unhandled duplicate test**](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-36985965a5a1d2da155f16ca9818ff0c4fae7f163663b4ee42f2e1941cbca43bR78-R97): The new integration test `test_duplicate_ingestion_returns_409` is left as a `pass`, so the duplicate-detection edge case is not actually verified.
> 
> Referred Code
> ```python
> """
> Ingest text content into the memory system.
> 
> This endpoint processes text content, chunks it if necessary,
> and stores it in the appropriate memOS.as memory tiers.
> 
> CRITICAL: Raw data is persisted to PostgreSQL BEFORE any processing
> to ensure 100% data retention (TN-CORE-101).
> 
> Args:
>     request: Ingestion request with content and metadata
>     background_tasks: FastAPI background tasks for async processing
>     db: Synchronous database session for raw persistence
>     memos_client: memOS.as client dependency
> 
> Returns:
>     IngestionResponse: Processing status and results
> 
> Raises:
>     HTTPException: On validation or processing errors
> ```
> 
> > Learn more about managing compliance [generic rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#configuration-options) or creating your own [custom rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#custom-compliance)
> 
> **Generic: Secure Error Handling**
> 
> **Objective:** To prevent the leakage of sensitive system information through error messages while providing sufficient detail for internal debugging.
> 
> **Status:** [**Exception leaked to client**](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-36985965a5a1d2da155f16ca9818ff0c4fae7f163663b4ee42f2e1941cbca43bR471-R475): The new `ingest_file` handler returns `HTTPException(..., detail=str(e))`, potentially exposing internal error details to end users.
> 
> Referred Code
> ```python
> except HTTPException:
>     raise
> except Exception as e:
>     logger.error(f"File ingestion failed: {e}")
>     raise HTTPException(status_code=500, detail=str(e))
> ```
> 
> > Learn more about managing compliance [generic rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#configuration-options) or creating your own [custom rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#custom-compliance)
> 
> ⚪	
> **Generic: Comprehensive Audit Trails**
> 
> **Objective:** To create a detailed and reliable record of critical system actions for security analysis and compliance.
> 
> **Status:** [**Missing user context**](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-36985965a5a1d2da155f16ca9818ff0c4fae7f163663b4ee42f2e1941cbca43bR123-R138): New ingestion persistence logs record the `ingestion_id` but do not include any user identifier or request principal context, making it hard to reconstruct who performed the action.
> 
> Referred Code
> ```python
>     logger.info(f"Raw ingestion persisted: {ingestion_id}")
> 
> except IntegrityError as e:
>     db.rollback()
>     logger.warning(f"Duplicate ingestion ID: {ingestion_id}")
>     raise HTTPException(
>         status_code=409,
>         detail=f"Ingestion {ingestion_id} already exists"
>     )
> except Exception as e:
>     db.rollback()
>     logger.error(f"Failed to persist raw ingestion: {e}")
>     raise HTTPException(
>         status_code=500,
>         detail="Failed to persist raw ingestion data"
>     )
> ```
> 
> > Learn more about managing compliance [generic rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#configuration-options) or creating your own [custom rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#custom-compliance)
> 
> **Generic: Secure Logging Practices**
> 
> **Objective:** To ensure logs are useful for debugging and auditing without exposing sensitive information like PII, PHI, or cardholder data.
> 
> **Status:** [**Potential sensitive paths**](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-87bf74ee0fb4087c78f25ecd0ec2a679e77c5617790e47729712c1e207af9475R107-R108): New logging includes `request.source_path` which may contain sensitive local filesystem information and should be assessed/redacted based on deployment context.
> 
> Referred Code
> ```python
> logger.info(f"Raw repository ingestion persisted: {ingestion_id} ({request.source_path})")
> ```
> 
> > Learn more about managing compliance [generic rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#configuration-options) or creating your own [custom rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#custom-compliance)
> 
> **Generic: Security-First Input Validation and Data Handling**
> 
> **Objective:** Ensure all data inputs are validated, sanitized, and handled securely to prevent vulnerabilities
> 
> **Status:** [**Raw persisted pre-validation**](https://github.com/ApexSigma-Solutions/InGest-LLM.as/pull/147/files#diff-36985965a5a1d2da155f16ca9818ff0c4fae7f163663b4ee42f2e1941cbca43bR103-R138): The new raw persistence writes `request.model_dump(...)` to the database before any explicit size/content validation in this function, which may allow storing excessively large or unsafe payloads depending on upstream validation behavior.
> 
> Referred Code
> ```python
> # STEP 1: IMMEDIATE RAW PERSISTENCE (TN-CORE-101)
> # Write raw data to PostgreSQL BEFORE any processing to ensure data retention
> try:
>     raw_record = RawIngestion(
>         ingestion_id=ingestion_id,
>         source_type="text",
>         content_type=request.metadata.content_type.value,
>         raw_payload=request.model_dump(mode="json"),
>         raw_metadata={
>             "source": request.metadata.source.value,
>             "tags": request.metadata.tags,
>             "source_url": request.metadata.source_url,
>             "title": request.metadata.title,
>         },
>         captured_at=datetime.utcnow(),
>         processed=False,
>     )
>     db.add(raw_record)
>     db.commit()
> 
>     logger.info(f"Raw ingestion persisted: {ingestion_id}")
> 
> 
>  ... (clipped 15 lines)
> ```
> 
> > Learn more about managing compliance [generic rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#configuration-options) or creating your own [custom rules](https://qodo-merge-docs.qodo.ai/tools/compliance/#custom-compliance)
> 
> Compliance status legend
> 🟢 - Fully Compliant
> 🟡 - Partial Compliant
> 🔴 - Not Compliant
> ⚪ - Requires Further Human Verification
> 🏷️ - Compliance label

@copilot please adress these issues

File: tests/verify/test_tn_core_103.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                        source_id VARCHAR(255) UNIQUE NOT NULL,
+                        platform VARCHAR(50),
+                        raw_payload JSONB NOT NULL,
+                        captured_at TIMESTAMP NOT NULL DEFAULT NOW(),
+                        processed BOOLEAN NOT NULL DEFAULT FALSE,
+                        processed_at TIMESTAMP,
+                        processing_attempts INTEGER NOT NULL DEFAULT 0,
+                        last_error TEXT
+                    )
+                """)
```
[the remark this answers is not in the document]
Comment (SteynSean11) [reply tests/verify/test_tn_core_103.py:47]: @copilot please correct the schema mismatch

File: tests/integration/test_raw_persistence.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            
+            raw_record = test_db_session.query(RawIngestion).filter_by(
+                ingestion_id=ingestion_id
+            ).first()
+            
+            assert raw_record is not None
+            assert raw_record.source_type == "python-repo"
+            assert raw_record.content_type == "repository"
+            assert raw_record.metadata["source_path"] == "/tmp/test-repo"
+            assert raw_record.metadata["max_files"] == 100
```
[the remark this answers is not in the document]
Comment (SteynSean11) [reply tests/integration/test_raw_persistence.py]: @copilot please address the metadata issue

File: src/ingest_llm_as/api/repository.py [hunk 3]
```diff
@@ -71,6 +81,45 @@ async def ingest_python_repository(
...
+            raw_payload=request.model_dump(mode="json"),
+            raw_metadata={
+                "repository_source": request.repository_source.value,
+                "source_path": request.source_path,
+                "max_files": request.max_files,
+                "max_file_size": request.max_file_size,
+                "include_patterns": request.include_patterns,
+                "exclude_patterns": request.exclude_patterns[:5],  # First 5 for brevity
+            },
+            captured_at=datetime.utcnow(),
```
[the remark this answers is not in the document]
Comment (SteynSean11) [reply src/ingest_llm_as/api/repository.py]: @copilot please change to timezone aware UTC format

File: src/ingest_llm_as/api/ingestion.py [hunk 4]
```diff
@@ -89,6 +99,43 @@ async def ingest_text(
...
+            status_code=409,
+            detail=f"Ingestion {ingestion_id} already exists"
+        )
+    except Exception as e:
+        db.rollback()
+        logger.error(f"Failed to persist raw ingestion: {e}")
+        raise HTTPException(
+            status_code=500,
+            detail="Failed to persist raw ingestion data"
+        )
```
[the remark this answers is not in the document]
Comment (SteynSean11) [reply src/ingest_llm_as/api/ingestion.py]: @copilot please adopt these changes

File: alembic/versions/1c6ce4b88374_create_raw_ingestions.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        postgresql_where=sa.text('processed = FALSE')
+    )
+    op.create_index('idx_raw_ingestions_captured_at', 'raw_ingestions', ['captured_at'])
+    op.create_index('idx_raw_ingestions_source_type', 'raw_ingestions', ['source_type'])
+    op.create_index('idx_raw_ingestions_ingestion_id', 'raw_ingestions', ['ingestion_id'], unique=True)
+
+
+def downgrade() -> None:
+    """Drop raw_ingestions table."""
+    op.drop_table('raw_ingestions')
```
[the remark this answers is not in the document]
Comment (SteynSean11) [reply alembic/versions/1c6ce4b88374_create_raw_ingestions.py:58]: @copilot please explicitly drop the indexes

File: src/ingest_llm_as/processors/conversation_ingestor.py [hunk 6]
```diff
@@ -34,17 +37,18 @@ async def start(self):
...
         self.running = False
-        self.neo4j_service.close()
+        await self.omegakg_client.close()
 
     async def process_pending_conversations(self) -> int:
         """
         Fetch and process unprocessed conversations.
         """
+        # asyncpg connection for polling
         conn = await asyncpg.connect(self.settings.raw_db_url.replace("postgresql+asyncpg", "postgresql"))
```
[the remark this answers is not in the document]
Comment (SteynSean11) [reply src/ingest_llm_as/processors/conversation_ingestor.py]: @copilot this is another suggestion to unify the raw_ingestion table with the raw_conversation table and the respective workers
